sklearn/metrics/_pairwise_distances_reduction/_argkminlabels.pyx

```diff
@@ -0,0 +1,198 @@
+from cython cimport floating, integral
+from cython.parallel cimport parallel, prange
+from libcpp.map cimport map as cmap
+from libc.stdlib cimport free
+
+cimport numpy as cnp
+
+cnp.import_array()
+
+from ._argkmin cimport PairwiseDistancesArgKmin64
+from ._datasets_pair cimport DatasetsPair64
+from ...utils._typedefs cimport ITYPE_t, DTYPE_t
+from ...utils._typedefs import ITYPE, DTYPE
+from ...utils._sorting cimport simultaneous_sort
+import numpy as np
+
+from sklearn.utils.fixes import threadpool_limits
+
+
+cpdef enum WeightingStrategy:
+    uniform = 0
+    distance = 1
+    other = 2
+
+cdef extern from *:
+    cdef cppclass pair "std::pair" [T, U]:
+        pair(T&, U&) except +
+    cdef cppclass map "std::map" [T, U]:
+        void insert(pair[T, U]&) except +
```

Comment:
Was there an issue with using the `map.insert` defined in Cython?

https://github.com/cython/cython/blob/278d07603e9ae9ce167f489c3c9855cb424cd5e7/Cython/Includes/libcpp/map.pxd#L106

Reply:
Nope, good catch